refactor(icons): replace fa icons EE-4459

testA113 opened the pull request:
Closes EE-4459

testA113 committed:
refactor(icons): remove fontawesome EE-4459

testA113 committed:
refactor(icons): fix typos EE-4459

testA113 committed:
refactor(icons): remove more feather props

chiptus commented on extensions.json:
```diff
@@ -10,7 +10,7 @@
     "Version": "1.0.1",
```
this file can be deleted

chiptus commented on app/assets/css/app.css:
```diff
@@ -827,3 +827,24 @@ input[style*='background-image: url("data:image/png'] {
   display: none;
 }
+
+.spin {
```
why not just use `animate-spin` tailwind class?
https://tailwindcss.com/docs/animation

testA113 replied:
tailwinds default spin duration is 1 second, while we were using a spin duration of 2s. 

I did add a spin slow animation to the tailwind config though, so will remove this and add use `animate-spin-slow`

testA113 replied:
tailwinds default spin is 1s, while portainer is 2s, I have made a custom tailwind definition for this in the tailwind config and only used this (so can now remove .spin)

chiptus commented on app/docker/views/services/create/createservice.html:
```diff
@@ -63,7 +63,7 @@
                 <!-- !host-port -->
                 <span style="margin: 0 10px 0 10px">
-                  <i class="fa fa-long-arrow-alt-right" aria-hidden="true"></i>
+                  <pr-icon icon="'svg-arrowright'"></pr-icon>
```
any reason not to use feather's arrow-right icon?

testA113 replied:
I got it as the replacement from https://portainer-style-guide.webflow.io/

![image](https://user-images.githubusercontent.com/83188384/198133870-9a3f9821-f8b9-4487-9e04-bbc2c5bf2aa9.png)

I'll rename this svg to arrowrightlong to be more obvious about how it is different from feather

chiptus commented on app/edge/EdgeDevices/EdgeDevicesView/AMTDevicesDatatable/columns/power-state.tsx:
```diff
@@ -48,7 +50,9 @@ export function PowerStateCell({
       </span>
-      <span>{isLoading && <i className="fa fa-cog fa-spin space-left" />}</span>
+      <span>
+        {isLoading && <Icon icon="settings" className="spin space-left" />}
```
in react, it's better to import the icon instead of using the string version. this will make sure in the future that we know exactly which icons are used, so we can drop the `import * from 'feather-icon` in the Icon component.

I mean like this:
```tsx
import {Settings} from 'react-feather'

<Icon icon={Settings} />
```

chiptus commented on app/edge/EdgeDevices/WaitingRoomView/Datatable/Datatable.tsx:
```diff
@@ -109,7 +109,7 @@ export function DataTable({
       <div className="col-sm-12">
         <Table.Container>
-          <Table.Title label="Edge Devices Waiting Room" icon="">
+          <Table.Title label="Edge Devices Waiting Room" icon="box">
```
`icon={Box}`

chiptus commented on app/kubernetes/components/kubectl-shell/kubectl-shell.controller.js:
```diff
@@ -22,7 +22,7 @@ export default class KubectlShellController {
   disconnect() {
```
this file can be deleted

testA113 replied:
deleted

chiptus commented on app/portainer/home/EdgeLoadingSpinner.tsx:
```diff
@@ -1,12 +1,14 @@
     <div className={clsx('row', styles.root)}>
       Connecting to the Edge environment...
-      <i className="fa fa-cog fa-spin space-left" />
+      <Icon icon="settings" className="spin space-left" />
```
```suggestion
      <Icon icon={Settings} className="spin space-left" />
```

chiptus replied:
still need to be updated

chiptus commented on app/portainer/filters/filters.js:
```diff
@@ -95,13 +95,13 @@ export function endpointTypeName(type) {
 export function environmentTypeIcon(type) {
   if (type === 3) {
-    return 'fab fa-microsoft';
+    return 'grid';
```
since this function is used by react components, we can directly import the icons:
```
import {Grid, Cloud} from 'react-feather'
import kubernetes2 from '@/assets/ico/vendor/kubernetes2.svg?c';
import dockericon from '@/assets/ico/vendor/dockericon.svg?c';
```

then if the last two are only used here, I'd suggest removing them from the Svg.tsx file

testA113 replied:
updated

chiptus commented on app/portainer/home/EnvironmentList/EnvironmentList.module.css:
```diff
@@ -48,7 +48,7 @@
 
   border: 0px;
-  padding: 5px 10px;
+  padding: 10px 10px;
```
```suggestion
  padding: 10px;
```

testA113 replied:
updated

chiptus commented on app/portainer/models/access.js:
```diff
@@ -6,7 +6,7 @@ export function UserAccessViewModel(data) {
   this.Override = false;
   this.Role = { Id: 0, Name: '-' };
-  this.icon = '<i class="fa fa-user" aria-hidden="true"></i>';
+  this.icon = '<pr-icon icon="\'user\'" ></pr-icon>';
```
is this actually working? is this being used somewhere?

testA113 replied:
the icon field isn't - removed it

chiptus commented on app/portainer/models/endpoint/formValues.js:
```diff
@@ -28,12 +28,7 @@ class PortainerEndpointInitFormValueEndpointSection {
-    'Manage the local Kubernetes environment'
-  ),
-  new PortainerEndpointInitFormValueEndpointSection(PortainerEndpointConnectionTypes.AGENT, 'Agent', 'fa fa-bolt', 'Connect to a Portainer agent'),
+  new PortainerEndpointInitFormValueEndpointSection(PortainerEndpointConnectionTypes.DOCKER_LOCAL, 'Docker', 'svg-dockericon', 'Manage the local Docker environment'),
```
import icons directly

testA113 replied:
This is only used by initEndpointController.js

Is this still used now that we've switched to the wizard?

testA113 replied:
I see your initEndpoint route comment, removed this

chiptus commented on app/portainer/rbac/components/roles-datatable/roles-datatable.html:
```diff
@@ -1,12 +1,12 @@
-        </div>
-        <div class="searchBar vertical-center">
-          <pr-icon icon="'search'" class="vertical-center" feather="true"></pr-icon>
+      <rd-widget-header icon="svg-filecode">
```
all tables are using class="toolBar". is there a reason why to replace it here?

testA113 replied:
The styling in the title looked off, so I made a quick change in this pr - updated to use class="toolBar" though

chiptus commented on app/portainer/views/init/admin/initAdmin.html:
```diff
@@ -54,16 +54,10 @@
-                      aria-hidden="true"
-                    ></i
-                  ></span>
+                  <span class="input-group-addon !rounded-r-[5px]">
```
just regular rounded-r or rounded-r-md won't work? why do we need that?

testA113 replied:
There's a conflicting style that used to give completely square corners, and the 5px is to match the `.input-group-addon` rounded corners. 

It could be worth playing around with the selector for this conflicting style, but I didn't want to get too far out of scope for this ticket.

chiptus commented on app/portainer/views/init/endpoint/initEndpoint.html:
```diff
@@ -28,7 +28,9 @@
                     <input type="radio" id="{{ type.Id }}" ng-model="ctrl.formValues.ConnectionType" ng-value="type.Value" />
```
this file can be removed

chiptus replied:
and the whole route

testA113 replied:
the whole `views/init/endpoint` directory?

testA113 replied:
removed views`/init/endpoint` in one commit so that it's easy to change if needed

chiptus commented on app/react/azure/DashboardView/DashboardView.tsx:
```diff
@@ -35,7 +33,7 @@ export function DashboardView() {
             <DashboardItem
               value={subscriptionsCount as number}
-              icon={SubscriptionsIcon}
+              icon="svg-subscription"
```
is it being used somewhere else?

chiptus replied:
anyway, in react, we should use direct import

testA113 replied:
updated

chiptus commented on app/react/azure/container-instances/CreateView/CreateContainerInstanceForm.tsx:
```diff
@@ -195,7 +196,7 @@ export function CreateContainerInstanceForm() {
                 loadingText="Deployment in progress..."
               >
-                <i className="fa fa-plus space-right" aria-hidden="true" />
+                <Icon icon="plus" className="mr-1" />
```
```suggestion
                <Icon icon={Plus} className="mr-1" />
```

chiptus commented on app/react/azure/container-instances/ListView/columns/ports.tsx:
```diff
@@ -26,8 +28,8 @@ function PortsCell({
     <a className="image-tag" href={`http://${ip}:${port.host}`} key={port.host}>
-      <i className="fa fa-external-link-alt" aria-hidden="true" /> {ip}:
-      {port.host}
+      <Icon icon="external-link" className="mr-1" />
```
```suggestion
      <Icon icon={ExternalLink} className="mr-1" />
```

chiptus commented on app/react/components/BoxSelector/BoxSelector.stories.tsx:
```diff
@@ -21,14 +21,14 @@ function Example() {
     {
       description: 'description 1',
-      icon: 'fa fa-rocket',
+      icon: 'svg-rocket',
```
use direct imports

chiptus commented on app/react/components/BoxSelector/BoxSelector.test.tsx:
```diff
@@ -23,14 +23,14 @@ test('should render with the initial value selected and call onChange when click
     {
       description: 'description 1',
-      icon: 'fa fa-rocket',
+      icon: 'svg-rocket',
```
use direct imports

chiptus commented on app/react/components/BoxSelector/BoxSelectorItem.stories.tsx:
```diff
@@ -11,7 +11,7 @@ const meta: Meta = {
     selected: false,
     description: 'description',
-    icon: 'fa-rocket',
+    icon: 'svg-rocket',
```
use direct imports

chiptus commented on app/react/components/Code/Code.tsx:
```diff
@@ -19,12 +18,10 @@ export function Code({ children, showCopyButton }: Props) {
-            )}
-            aria-hidden="true"
+          <Icon
+            icon={copiedSuccessfully ? 'check' : 'copy'}
```
```suggestion
            icon={copiedSuccessfully ? Check : Copy}
```

chiptus replied:
this can still be fixed

testA113 replied:
updated

chiptus commented on app/react/components/DashboardItem/DashboardItem.stories.tsx:
```diff
@@ -23,21 +23,21 @@ function Template({ value, icon, type }: StoryProps) {
 Primary.args = {
   value: 1,
-  icon: 'fa fa-th-list',
+  icon: 'list',
```
use direct imports

chiptus replied:
also here

testA113 replied:
updated

chiptus commented on app/react/components/DashboardItem/DashboardItem.test.tsx:
```diff
@@ -11,19 +11,19 @@ test('should show provided resource value', async () => {
 
 test('should show provided resource type', async () => {
-  const { getByLabelText } = renderComponent(0, '', 'Test');
+  const { getByLabelText } = renderComponent(0, 'user', 'Test');
```
use direct imports

testA113 replied:
updated

chiptus commented on app/react/components/DashboardItem/DashboardItem.tsx:
```diff
@@ -35,7 +29,7 @@ export function DashboardItem({
           )}
         >
-          <Icon icon={icon} feather={featherIcon} className="feather" />
+          <Icon icon={icon} className="feather" />
```
```suggestion
          <Icon icon={icon} />
```

testA113 replied:
updated

chiptus commented on app/react/components/Icon.tsx:
```diff
@@ -57,27 +69,25 @@ export function Icon({ icon, feather, className, mode, size }: Props) {
+    return (
+      <Svg
+        icon={'placeholder' as keyof typeof SvgIcons}
+        className={clsx(classes, inline ? '!inline' : '!flex')}
```
inline is only for the placeholder?

chiptus replied:
I actually think you should throw an error here. this way the developer would notice using a wrong icon. when do we expect to need to use the placeholder?

testA113 replied:
Inline as a prop is passed over from EE - I updated the SVG component to match EE too
```
  if (icon.indexOf('svg-') === 0) {
    const svgIcon = icon.replace('svg-', '');
    return (
      <Svg
        icon={svgIcon as keyof typeof SvgIcons}
        className={clsx(classes, inline ? '!inline' : '!flex')}
        aria-label={ariaLabel}
      />
    );
  }
```

chiptus commented on app/react/components/InformationPanel.tsx:
```diff
@@ -33,7 +34,8 @@ export function InformationPanel({
                     >
                       <Button color="link" onClick={() => onDismiss()}>
-                        <i className="fa fa-times" /> dismiss
+                        <Icon icon="x" />
```
use the icon property for the button.

so 
```
<Button icon={X} color="link" onClick={() => onDismiss()}>
```

chiptus replied:
still relevant

testA113 replied:
updated

chiptus commented on app/react/components/PasswordCheckHint.tsx:
```diff
@@ -17,12 +17,12 @@ export function PasswordCheckHint({
         The password must be at least {minPasswordLength} characters long.
         {passwordValid && (
-          <i className="fa fa-check green-icon space-left" aria-hidden="true" />
+          <Icon icon="check" className="!ml-1" mode="success" />
```
```suggestion
          <Icon icon={Check} className="!ml-1" mode="success" />
```

chiptus commented on app/react/components/ViewLoading/ViewLoading.tsx:
```diff
@@ -18,7 +20,7 @@ export function ViewLoading({ message }: Props) {
         <span className={styles.message}>
           {message}
-          <i className="fa fa-cog fa-spin space-left" />
+          <Icon icon="settings" className="spin ml-1" />
```
```suggestion
          <Icon icon={Settings} className="spin ml-1" />
```

chiptus commented on app/react/components/Widget/Widget.stories.tsx:
```diff
@@ -20,7 +20,7 @@ const meta: Meta<WidgetProps> = {
     loading: false,
     title: 'Title',
-    icon: 'fa-rocket',
+    icon: 'svg-rocket',
```
direct import

chiptus commented on app/react/components/Widget/WidgetTitle.tsx:
```diff
@@ -26,11 +24,7 @@ export function WidgetTitle({
-              feather={featherIcon}
-              className="space-right feather"
-            />
+            <Icon icon={icon} className="space-right feather" />
```
can we remove the feather class?

testA113 replied:
Yes! :D I'm doing this work in the Feather -> Lucide ticket - EE-4472 - completely removing this feather class

chiptus replied:
I just meant that we might be able to remove this use

testA113 replied:
updated

chiptus commented on app/react/components/buttons/ButtonGroup.stories.tsx:
```diff
@@ -15,27 +17,23 @@ function Template({
     <ButtonGroup size={size}>
       <Button color="primary" onClick={() => {}}>
-        <i className="fa fa-play space-right" aria-hidden="true" />
+        <Icon icon="play" />
```
use direct import

chiptus commented on app/react/components/buttons/LoadingButton.test.tsx:
```diff
@@ -15,9 +15,6 @@ test('when isLoading is true should show spinner and loading text', async () =>
   const buttonLabel = queryByText(children);
   expect(buttonLabel).toBeNull();
 
-  const spinner = await findByLabelText('loading');
```
why remove?

testA113 replied:
The svg label doesn't seem to show in the tests, although it shows in the loaded pages outside of testing 
![image](https://user-images.githubusercontent.com/83188384/198238778-38197bb0-df3b-46a0-babe-a5779c4bb294.png)

chiptus commented on app/react/components/buttons/LoadingButton.test.tsx:
```diff
@@ -32,9 +29,6 @@ test('should show children when false', async () => {
     </LoadingButton>
   );
 
-  const buttonLabel = queryByText(children);
```
how do you check that the children are there?

testA113 replied:
Added this back, sorry for removing

chiptus commented on app/react/components/datatables/ActionsMenu.tsx:
```diff
@@ -19,7 +21,7 @@ export function ActionsMenu({ children }: Props) {
             )}
           >
-            <i className="fa fa-ellipsis-v" aria-hidden="true" />
+            <Icon icon="more-vertical" />
```
use direct import

chiptus commented on app/react/components/datatables/ExpandingCell.tsx:
```diff
@@ -15,22 +17,15 @@ export function ExpandingCell<
             {...row.getToggleRowExpandedProps()}
-            className={`fas ${arrowClass(row.isExpanded)} space-right`}
-            aria-hidden="true"
+            icon={row.isExpanded ? 'chevron-down' : 'chevron-right'}
```
use direct import

testA113 replied:
nice spot - updated

chiptus commented on app/react/components/datatables/Filter.tsx:
```diff
@@ -28,14 +30,7 @@ export function MultipleSelectionFilter({
-            )}
-            aria-hidden="true"
-          />
+          <Icon icon={enabled ? 'check' : 'filter'} className="ml-1" />
```
use direct import

chiptus commented on app/react/components/datatables/SortbySelector.tsx:
```diff
@@ -50,15 +39,18 @@ export function SortbySelector({
       <div className={styles.sortByElement}>
         <button
-          className={styles.sortButton}
+          className={clsx(styles.sortButton, 'h-[34px]')}
```
why not just `h-9`?

testA113 replied:
It has to match the exact height of the buttons it's beside in the home dashboard
![image](https://user-images.githubusercontent.com/83188384/198241132-7e038e25-66be-49c6-8171-9c493275c138.png)

chiptus commented on app/react/components/datatables/SortbySelector.tsx:
```diff
@@ -50,15 +39,18 @@ export function SortbySelector({
+          className={clsx(styles.sortButton, 'h-[34px]')}
           type="button"
-          disabled={!sortByButton}
+          disabled={!sortByButton || !value}
```
this and line 51 are opposites, right? maybe keep that as a variable (`const sorted = sortByButton && !!value`)?

testA113 replied:
nice, updated

chiptus commented on app/react/components/datatables/TableSettingsMenuAutoRefresh.tsx:
```diff
@@ -44,14 +45,9 @@ export function TableSettingsMenuAutoRefresh({ onChange, value }: Props) {
               styles.check
             )}
-            onTransitionEnd={() => setIsCheckVisible(false)}
+            // onTransitionEnd={() => setIsCheckVisible(false)}
```
why is this hidden?

testA113 replied:
thank-you for finding this! - un-commented

chiptus commented on app/react/components/datatables/TableSettingsMenuAutoRefresh.tsx:
```diff
@@ -44,14 +45,9 @@ export function TableSettingsMenuAutoRefresh({ onChange, value }: Props) {
-              aria-hidden="true"
-              style={{ marginTop: '7px' }}
-            />
+            <Icon icon="check" className="!ml-1" mode="success" />
```
use direct import

chiptus commented on app/react/docker/DashboardView/ContainerStatus.tsx:
```diff
@@ -17,37 +17,21 @@ export function ContainerStatus({ containers }: Props) {
-            className={clsx('icon icon-sm icon-success')}
-            feather
-          />
+          <Icon icon="power" className={clsx('icon icon-sm icon-success')} />
```
looks like clsx is not needed in this file

testA113 replied:
updated

chiptus commented on app/react/docker/app-templates/TemplateListSort/TemplateListSort.tsx:
```diff
@@ -1,3 +1,7 @@
+import clsx from 'clsx';
```
what's the difference of this file from `app/react/components/datatables/SortbySelector.tsx`

testA113 replied:
It seems like the only difference is type of the filter options and values, not sure about the styles. Will leave this outside the scope of this ticket though as it could be worth making a storybook story for this too

chiptus commented on app/react/docker/containers/ListView/ContainersDatatable/columns/ports.tsx:
```diff
@@ -37,7 +39,7 @@ function PortsCell({ value: ports }: Props) {
       rel="noreferrer"
     >
-      <i className="fa fa-external-link-alt" aria-hidden="true" />
+      <Icon icon="external-link" />
```
direct imports

chiptus commented on app/react/docker/stacks/ItemView/StackContainersDatatable.tsx:
```diff
@@ -55,7 +55,7 @@ export function StackContainersDatatable({ environment, stackName }: Props) {
       <Datatable
         titleOptions={{
-          icon: 'fa-cubes',
+          icon: 'svg-cubes',
```
direct imports

chiptus commented on app/react/edge/components/EdgeScriptForm/OsSelector.tsx:
```diff
@@ -20,7 +21,7 @@ export function OsSelector({ onChange, value }: Props) {
               label: (
                 <>
-                  <i className="fab fa-linux space-right" aria-hidden="true" />
+                  <Icon icon="svg-linux" className="mr-1" />
```
direct imports

chiptus commented on app/react/portainer/access-control/AccessControlPanel/AccessControlPanelDetails.tsx:
```diff
@@ -169,7 +170,7 @@ function InheritanceMessage({
     <tr>
       <td colSpan={2} aria-label="inheritance-message">
-        <i className="fa fa-info-circle space-right" aria-hidden="true" />
+        <Icon icon="info" mode="primary" className="mr-1" />
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/EnvironmentTypeSelectView/environment-types.ts:
```diff
@@ -6,36 +6,36 @@ export const environmentTypes = [
     id: 'dockerStandalone',
     title: 'Docker Standalone',
-    icon: 'fab fa-docker',
+    icon: 'svg-dockericon',
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardAzure/WizardAzure.tsx:
```diff
@@ -38,7 +38,15 @@ const initialValues: FormValues = {
+const options = [
+  {
+    description: '',
+    icon: <BadgeIcon icon="svg-dataflow2" size="3xl" />,
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardDocker/WizardDocker.tsx:
```diff
@@ -23,28 +24,28 @@ const defaultOptions: BoxSelectorOption<
   {
     id: 'agent',
-    icon: 'svg-agent',
+    icon: <BadgeIcon icon="zap" size="3xl" />,
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardKubernetes/DeploymentScripts.tsx:
```diff
@@ -97,10 +98,7 @@ function DeployCode({
-            className="fa fa-info-circle blue-icon space-right"
-            aria-hidden="true"
-          />
+          <Icon icon="info" mode="primary" className="mr-1" />
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardKubernetes/KubeConfigTeaserForm.tsx:
```diff
@@ -85,7 +86,8 @@ export function KubeConfigTeaserForm() {
                 disabled
               >
-                <i className="fa fa-plug" aria-hidden="true" /> Connect
+                <Icon icon="svg-plug" />
```
use the icon prop of LoadingButton. and import the icon :)

testA113 replied:
The loading button  wasn't diplaying the icon passed as a prop (see the LoadingButtonIcon funtion). I updated the loading button so that it can display icons now

testA113 replied:
https://github.com/portainer/portainer/pull/7907/commits/9ba7bf0f2435274841a5c453146d343608eb36de

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardKubernetes/WizardKubernetes.tsx:
```diff
@@ -25,22 +26,22 @@ interface Props {
   {
     id: 'agent_endpoint',
-    icon: 'svg-agent',
+    icon: <BadgeIcon icon="zap" size="3xl" />,
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/HomeView/HomeView.tsx:
```diff
@@ -61,8 +61,8 @@ export function HomeView() {
                           localEnvironmentAdded.type === EnvironmentType.Docker
-                            ? 'fab fa-docker'
-                            : 'fas fa-dharmachakra'
+                            ? 'svg-dockericon'
```
direct imports

chiptus commented on app/react/portainer/environments/wizard/components/Option/Option.tsx:
```diff
@@ -44,11 +45,17 @@ export function Option({
+            icon={icon}
+            className={clsx(
+              'block',
+              icon === 'svg-dockericon' ? styles.iconLarge : styles.icon
```
this is too specific. I think in endpoint-types, just import the icon directly, then this L52-58 won't be needed at all.

chiptus reviewed (commented):
Great job! Literally. 

Please have a look at my comments

testA113 committed:
implement EE-4469

testA113 committed:
tailwind spin, svg arrow rename, rm old files

testA113 committed:
remove initEndpoint route

chiptus commented on app/edge/EdgeDevices/EdgeDevicesView/AMTDevicesDatatable/columns/power-state.tsx:
```diff
@@ -48,7 +50,11 @@ export function PowerStateCell({
-      <span>{isLoading && <i className="fa fa-cog fa-spin space-left" />}</span>
+      <span>
+        {isLoading && (
+          <Icon icon="settings" className="animate-spin-slow space-left" />
```
```suggestion
          <Icon icon={Settings} className="animate-spin-slow space-left" />
```

chiptus commented on app/react/azure/container-instances/CreateView/PortsMappingField.tsx:
```diff
@@ -82,7 +83,7 @@ function Item({
 
         <span className="mx-3">
-          <i className="fa fa-long-arrow-alt-right" aria-hidden="true" />
+          <Icon icon="svg-arrowrightlong" />
```
use direct import

chiptus commented on app/react/components/ViewLoading/ViewLoading.tsx:
```diff
@@ -18,7 +20,7 @@ export function ViewLoading({ message }: Props) {
         <span className={styles.message}>
           {message}
-          <i className="fa fa-cog fa-spin space-left" />
+          <Icon icon="settings" className="animate-spin-slow ml-1" />
```
```suggestion
          <Icon icon={Settings} className="animate-spin-slow ml-1" />
```

chiptus commented on app/react/edge/components/EdgeScriptForm/OsSelector.tsx:
```diff
@@ -29,10 +30,7 @@ export function OsSelector({ onChange, value }: Props) {
-                    className="fab fa-windows space-right"
-                    aria-hidden="true"
-                  />
+                  <Icon icon="grid" className="mr-1" />
```
```suggestion
                  <Icon icon={Grid} className="mr-1" />
```

chiptus commented on app/react/components/buttons/LoadingButton.tsx:
```diff
@@ -24,10 +26,10 @@ export function LoadingButton({
-            aria-label="loading"
-            aria-hidden="true"
+          <Icon
+            icon="svg-circlenotch"
```
use direct import. I think then you'll see the label in tests

chiptus commented on app/react/components/buttons/LoadingButton.stories.tsx:
```diff
@@ -15,7 +17,8 @@ interface Args {
   return (
     <LoadingButton loadingText={loadingText} isLoading={isLoading}>
-      <i className="fa fa-download" aria-hidden="true" /> Download
+      <Icon icon="download" />
```
use icon prop

testA113 replied:
updated

chiptus commented on app/react/components/buttons/LoadingButton.stories.tsx:
```diff
@@ -30,7 +33,8 @@ export const Example = Template.bind({});
   return (
     <LoadingButton loadingText="loading" isLoading>
-      <i className="fa fa-download" aria-hidden="true" /> Download
+      <Icon icon="download" />
```
use icon prop

testA113 replied:
updated

testA113 committed:
address comments

testA113 committed:
wip update import style
note that the feather class retains styling

testA113 committed:
small fixes

testA113 committed:
apply EE changes

testA113 committed:
fix recent changes

testA113 committed:
Component name to match EE

testA113 committed:
use icon for styles

testA113 committed:
remove feather

testA113 committed:
fix prettier errors

testA113 committed:
remove more svg strings EE-4459

chiptus commented on app/edge/EdgeDevices/EdgeDevicesView/EdgeDevicesDatatable/EdgeDevicesDatatableActions.tsx:
```diff
@@ -41,22 +42,20 @@ export function EdgeDevicesDatatableActions({
           onClick={() => onAssociateOpenAMTClick(selectedItems)}
-          icon="link"
-          featherIcon
+          icon={Link}
```
Link is imported from `@@/Link` because we have a component by that name. if we need both of them, then for the icon you can do `import {Link as LinkIcon} from 'react-feather'`

testA113 replied:
Great spotting! :) updated

chiptus commented on app/index.html:
```diff
@@ -62,9 +62,26 @@
+                <div class="ml-0 mr-0 flex items-center justify-center text-center">
                   Loading Portainer...
-                  <i class="fa fa-cog fa-spin" style="margin-left: 5px"></i>
+                  <!-- adding the svg inline means that the settings icon loads in time (vs pr-icon) and with the correct color (vs img) -->
```
a tech debt ticket would be to remove this all view from HTML and move it to a component. but that's an idea for the future

chiptus commented on app/react/components/Icon.tsx:
```diff
@@ -56,28 +59,23 @@ export function Icon({ icon, feather, className, mode, size }: Props) {
       <Svg
         icon={svgIcon as keyof typeof SvgIcons}
-        className={clsx(classes, '!flex')}
+        className={clsx(classes, inline ? '!inline' : '!flex')}
```
I'm still not sure where we need this. maybe just use `inline-flex`?

testA113 replied:
I agree, updated

chiptus commented on app/react/components/PageHeader/NotificationsMenu.tsx:
```diff
@@ -69,7 +70,7 @@ export function NotificationsMenu() {
           )}
         >
-          <Icon icon="bell" feather />
+          <Icon icon={Bell} className="feather" />
```
in other places (in this file), you didn't need the className. why is it needed for this one?

testA113 replied:
removed

chiptus commented on app/react/components/Widget/Widget.stories.tsx:
```diff
@@ -9,7 +11,7 @@ import { WidgetTaskbar } from './WidgetTaskbar';
   loading: boolean;
   title: string;
-  icon: string;
+  icon: string | ReactNode;
```
`string` is a `ReactNode`, so can replace this with 
```suggestion
  icon: ReactNode;
```

testA113 replied:
updated all occurrences of this

chiptus commented on app/react/components/Widget/Widget.stories.tsx:
```diff
@@ -49,16 +51,7 @@ function WidgetWithCustomImage({
-          />
-        }
-      />
+      <WidgetTitle title={title} icon={icon} />
```
actually, the whole meaning of this story is to use a custom image, otherwise it's just the same story

testA113 replied:
updated to the icon string arg as the custom image  
```
WidgetWithCustomImage.args = {
  icon: 'https://via.placeholder.com/150',
};
```

chiptus commented on app/react/components/buttons/LoadingButton.tsx:
```diff
@@ -21,19 +25,22 @@ export function LoadingButton({
   );
 }
+
+function LoadingButtonIcon(isLoading: boolean) {
```
should take care of the possibility that a default icon is passed. i.e we use
`<LoadingButton icon={Settings}>Submit</LoadingButton>`
```suggestion
function LoadingButtonIcon(isLoading: boolean, defaultIcon: ReactNode) {
```

testA113 replied:
updated

chiptus commented on app/react/kubernetes/ingresses/CreateIngressView/Annotations/index.tsx:
```diff
@@ -69,13 +70,12 @@ export function Annotations({
           </div>
           <div className="col-sm-3 !pl-0 !m-0">
-            <button
+            <Button
```
this might be a good time to remove the `className` and type props and just use `size="sm" color="dangerlight"`

just a suggestion

testA113 replied:
updated

chiptus commented on app/react/portainer/HomeView/EnvironmentList/EnvironmentItem/EnvironmentIcon.tsx:
```diff
@@ -17,7 +17,7 @@ export function EnvironmentIcon({ type }: Props) {
     case EnvironmentType.Docker:
       return (
-        <img src={docker} width="60" alt="azure endpoint" aria-hidden="true" />
+        <img src={docker} width="60" alt="docker endpoint" aria-hidden="true" />
```
good catch.

chiptus commented on app/react/portainer/HomeView/EnvironmentList/EnvironmentItem/EnvironmentStatsItem.tsx:
```diff
@@ -1,27 +1,25 @@
 interface Props extends IconProps {
   value: string | number;
-  icon: string;
+  icon: string | ReactNode;
```
```suggestion
  icon: ReactNode;
```

or even better (to support changes in IconProps):

```suggestion
  icon: IconProps['icon'];
```

testA113 replied:
updated

testA113 replied:
updated

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardAzure/WizardAzure.tsx:
```diff
@@ -38,7 +40,15 @@ const initialValues: FormValues = {
 };
 
-const options = [buildOption('api', 'svg-api', 'API', '', 'api')];
+const options = [
```
nice! just to make sure we don't miss required props. you can type the array
```suggestion
const options: Array<BoxSelectorItem<'api'>> = [
```

testA113 replied:
updated

chiptus commented on app/react/portainer/environments/wizard/EnvironmentsCreationView/WizardKubernetes/KubeConfigTeaserForm.tsx:
```diff
@@ -85,7 +88,8 @@ export function KubeConfigTeaserForm() {
                 disabled
               >
-                <i className="fa fa-plug" aria-hidden="true" /> Connect
+                <Icon icon={Plug} />
```
just pass it as a prop to LoadingButton

testA113 replied:
updated

chiptus reviewed (commented):
Great job! it's a very big PR, lots of small things. Very nice! I did find some stuff you should check

chiptus commented on app/react/portainer/environments/wizard/components/Option/Option.tsx:
```diff
@@ -44,11 +45,11 @@ export function Option({
-        {Icon ? (
-          <Icon selected={active} className={styles.iconComponent} />
+      <div className="flex items-end justify-center text-center mt-2">
+        {IconComponent ? (
+          <IconComponent selected={active} className={styles.iconComponent} />
         ) : (
-          <i className={clsx(icon, 'block', styles.icon)} />
+          <Icon icon={icon} className={clsx('block', styles.icon)} />
         )}
```
this snippet was needed because we had `KaaSIcon` and string icons. now all of them are components, so we can just do:
```ts
const IconComponent = icon;

<div className="flex items-end justify-center text-center mt-2">
    <IconComponent selected={active} className={styles.iconComponent} />
</div>
```

testA113 replied:
updated

chiptus commented:
@testA113 I added some comments, and went over my old ones to resolve those that were fixed. Please have a look

chiptus commented on app/docker/views/services/edit/service.html:
```diff
@@ -78,63 +78,76 @@
                 <span class="text-muted">{{ webhookURL | truncatelr }}</span>
                 <button type="button" class="btn btn-sm btn-primary btn-sm space-left" ng-if="webhookURL" ng-click="copyWebhook()">
-                  <span><i class="fa fa-copy space-right" aria-hidden="true"></i>Copy link</span>
+                  <span>
```
in EE this span was removed (to be expected as it is not needed). please remove it here too

testA113 replied:
updated

chiptus commented on app/kubernetes/components/datatables/applications-stacks-datatable/applicationsStacksDatatable.html:
```diff
@@ -178,7 +178,7 @@
           <td>
             <a class="vertical-center" ui-sref="kubernetes.stacks.stack.logs({ namespace: item.ResourcePool, name: item.Name })">
-              <pr-icon icon="'file-text'" size="'md'" title="Logs" feather="true"></pr-icon>
+              <pr-icon icon="'file-text'" size="'md'" title="Logs"></pr-icon>
```
EE doesn't have the size prop. which one is correct?

https://github.com/portainer/portainer-ee/pull/2472/files#diff-f28f79bb14461bf2d4822a175f8c8cbf110c88d97b880d5ddfb852cb578df2b2R182

testA113 replied:
removed the size, the default size (like in EE) is better

chiptus commented on app/kubernetes/components/datatables/configurations-datatable/configurationsDatatable.html:
```diff
@@ -90,7 +90,7 @@
         <div ng-if="$ctrl.isAdmin && !$ctrl.settings.showSystem" class="flex flex-row w-full">
           <span class="small text-muted mt-1 vertical-center">
-            <pr-icon icon="'info'" feather="true" mode="'primary'" class="vertical-center"></pr-icon>
+            <pr-icon icon="'info'" mode="'primary'" class="vertical-center"></pr-icon>
```
EE doesn't have the class prop. I guess it's not needed

https://github.com/portainer/portainer-ee/pull/2472/files#diff-06d7328e875e75f0fd2e75c8f4802cf7df33c0c9936a5456b5bdf4736cc15bc3R93

testA113 replied:
not needed now the Icon.tsx styles are updated - removed from CE

chiptus commented on app/kubernetes/views/applications/stats/stats.html:
```diff
@@ -62,13 +64,13 @@
               </div>
               <span>
-                <pr-icon id="refreshRateChange" icon="'check'" feather="true" mode="'success'" size="'sm'"></pr-icon>
+                <pr-icon id="refreshRateChange" icon="'check'" mode="'success'" size="'sm'"></pr-icon>
```
I'm pretty sure this should have `style="display: none"` like in EE https://github.com/portainer/portainer-ee/pull/2472/files#diff-ca2701e089fd3017c1463150929960dedf4c650b70e5944b61a7ad8602193691R67

testA113 replied:
updated

chiptus commented on app/kubernetes/views/deploy/deploy.html:
```diff
@@ -118,7 +118,7 @@
                       <span class="col-sm-12 text-muted small" ng-show="ctrl.state.DeployType === ctrl.ManifestDeployTypes.COMPOSE">
                         <p class="vertical-center">
-                          <pr-icon icon="'alert-circle'" mode="'warning'" feather="true"></pr-icon>
+                          <pr-icon icon="'alert-circle'" mode="'warning'"></pr-icon>
```
EE has `info` icon. https://github.com/portainer/portainer-ee/pull/2472/files#diff-3e3185001e467389031d8b15861e453bb2894bd8f24c293598613fd3eab687f9R121

testA113 replied:
Warning messages should have 
```icon="'alert-triangle'" mode="'warning'"```
as per https://www.figma.com/file/g5TUMngrblkXM7NHSyQsD1/New-UI?node-id=144%3A2236&t=URpyH2qJo8W6VoKn-0
I've updated cases of 
```icon="'alert-circle'" mode="'warning'"```
to match this in CE and EE. I won't make the additional changes around the text color as it should be covered in a separate ticket

chiptus commented on app/portainer/filters/filters.js:
```diff
@@ -95,13 +98,13 @@ export function endpointTypeName(type) {
 export function environmentTypeIcon(type) {
   if (type === 3) {
-    return 'fab fa-microsoft';
+    return MicrosoftIcon;
```
can you sync this function with the one on EE?

testA113 replied:
updated

chiptus replied:
this was reverted as well

testA113 replied:
Sorry, I included this in the reverted commit because I thought you wanted to decrease the scope

testA113 replied:
Updated bringing over the function from EE again

chiptus reviewed (changes requested):
found a few more things after comparing the CE and EE Prs

testA113 committed:
Loading button to correctly use icon

testA113 committed:
address comments - 1

testA113 committed:
loading button update

testA113 committed:
address comments - 2

testA113 committed:
address comments - 3

testA113 committed:
address comments - 4

testA113 committed:
correct icons for alerts

testA113 committed:
theme sensitive table icon

testA113 committed:
stack button group icon to match EE EE-4459

testA113 committed:
update loading button test

testA113 committed:
remaining fa removed

testA113 committed:
queryByRole

testA113 committed:
Merge branch 'develop' into
refactor/EE-4459/replace-fa-icons

chiptus commented on app/react/components/Widget/Widget.stories.tsx:
```diff
@@ -54,7 +56,7 @@ function WidgetWithCustomImage({
           <img
             className="custom-header-ico space-right"
-            src={icon}
+            src={icon as string}
```
just a note, it's usually better to check `typeof icon==='string'` and doing something if it's false, then assuming it is a string (with `as`)

testA113 replied:
updated

chiptus commented on app/react/components/buttons/LoadingButton.tsx:
```diff
@@ -21,19 +26,25 @@ export function LoadingButton({
   );
 }
+
+function LoadingButtonIcon(isLoading: boolean, defaultIcon: ReactNode) {
+  if (defaultIcon && !isLoading) {
+    return defaultIcon;
+  }
+  if (isLoading) {
+    return (
+      <Icon
+        icon={CircleNotch}
+        className="animate-spin-slow ml-1"
+        aria-label="loading"
+      />
+    );
+  }
+  return null;
+}
```
just a suggestion:
```suggestion
function loadingButtonIcon(isLoading: boolean, defaultIcon?: ReactNode) {
  if (!isLoading) {
    return defaultIcon;
  }
  
    return (
      <Icon
        icon={CircleNotch}
        className="animate-spin-slow ml-1"
        aria-label="loading"
      />
    );
  }
}
```

this is doing a fast return. if not loading just return the default icon (which can be undefined), the rest of the code happens when isLoading is true. then we return the spinner.

also notice I renamed the function name to lowerCaseCamelCase. because it's a regular and not a component it should start with lower case character

chiptus replied:
another option is not to use a function at all. create a spinner component for the icon. and then use `icon={isLoading : <Spinner /> : icon}`

testA113 replied:
nice suggestion for the loadingButtonIcon function - it looks a lot cleaner - updated

testA113 committed:
Revert "correct icons for alerts"
(includes reverting environmentTypeIcon)

This reverts commit 7770d4c538683e49fe11f92cf2b68f0563adeac2.

testA113 committed:
revert stack icon change

chiptus commented on app/react/components/BoxSelector/BoxSelectorItem.css:
```diff
@@ -13,8 +13,7 @@
 
-.boxselector_header .fa,
-.fab {
+.boxselector_header {
```
this css rule is related to icons. should be removed totally

testA113 replied:
updated

chiptus reviewed (approved):
LGTM, added some suggestions

testA113 committed:
address comments - 5

testA113 committed:
refactor(icons): replace feather icon with lucide [EE-4472] (#8059)

refactor(icon) replace feather with lucide EE-4472

substitute custom icons for lucide icons EE-4472

use file-code instead of broken svg EE-4469

testA113 committed:
Merge branch 'develop' into refactor/EE-4459/replace-fa-icons

testA113 committed:
Merge branch 'develop' into refactor/EE-4459/replace-fa-icons

testA113 committed:
Merge branch 'develop' into refactor/EE-4459/replace-fa-icons

testA113 committed:
styles for lucide wizard option
